src/Orckestra.Composer.Search/Services/BaseSearchViewService.cs

```diff
@@ -214,65 +202,22 @@ private IList<SearchSortBy> GetSearchSortByList(SearchType searchType)
-        {
-            var cultureInfo = createSearchViewModelParam.SearchParam.Criteria.CultureInfo;
-
-            string variantId = null;
-            if (productDocument.PropertyBag.ContainsKey(VariantPropertyBagKey))
-            {
-                variantId = productDocument.PropertyBag[VariantPropertyBagKey] as string;
-            }
-
-            var productSearchVm = ViewModelMapper.MapTo<ProductSearchViewModel>(productDocument, cultureInfo);
-            productSearchVm.ProductId = productDocument.ProductId;
-            productSearchVm.BrandId = ExtractLookupId("Brand_Facet", productDocument.PropertyBag);
-            MapProductSearchViewModelInfos(productSearchVm, productDocument, cultureInfo);
-            MapProductSearchViewModelUrl(productSearchVm, variantId, cultureInfo, createSearchViewModelParam.SearchParam.Criteria.BaseUrl);
-            MapProductSearchViewModelImage(productSearchVm, imgDictionary);
-            productSearchVm.IsAvailableToSell = await GetProductSearchViewModelAvailableForSell(productSearchVm, productDocument).ConfigureAwait(false);
-            var pricing = await PriceProvider.GetPriceAsync(productSearchVm.HasVariants, productDocument).ConfigureAwait(false);
-            MapProductSearchViewModelPricing(productSearchVm, pricing);
-            productSearchVm.IsRecurringOrderEligible = RecurringOrdersSettings.Enabled && productDocument.PropertyBag.IsRecurringOrderEligible();
-
-            productSearchVm.Context["IsRecurringOrderEligible "] = productSearchVm.IsRecurringOrderEligible;
-
-            return productSearchVm;
-        }
-
-        protected virtual string ExtractLookupId(string fieldName, PropertyBag propertyBag)
-        {
-            if (propertyBag == null) { return null; }
-            var fieldValue = propertyBag.ContainsKey(fieldName) 
-                ? (propertyBag[fieldName] as string ?? (propertyBag[fieldName] as string[])[0])
-                : null;
-            if (string.IsNullOrWhiteSpace(fieldValue)) { return null; }
-
-            var extractedValues = fieldValue.Split(new[] { "::" }, StringSplitOptions.None);
-            return extractedValues.Length < 3
-                ? null
-                : extractedValues[2];
-        }
 
-        protected virtual async Task<bool> GetProductSearchViewModelAvailableForSell(
-            ProductSearchViewModel productSearchViewModel,
-            ProductDocument productDocument)
+        // NOTE: when fetching data for products from OCC APIs, make sure to query data in batches for optimal performance
+        // https://docs.orckestra.com/developer-documentation/platform-performance/batch-api-requests
+        protected virtual async Task<IList<ProductSearchViewModel>> AppendProductSearchViewModels(IList<(ProductSearchViewModel, ProductDocument)> productSearchResultList)
```

Comment:
Append sounds like "adding new elements to a list", but here we add data to existing view models, rather than adding new ones. I suggest the term "enrich":

**Enrich**AppendProductSearchViewModels 

![image](https://user-images.githubusercontent.com/1391705/141774654-6d3b4b03-09cb-48f5-80b5-3707e46e4378.png)